app/controllers/assessments_controller.rb

```diff
@@ -5,78 +5,74 @@ class AssessmentsController < ApplicationController
-        assessment_placeholder = assessment_placeholder.merge(params.permit(:threshold_hash).to_h)
-        assessment_placeholder = assessment_placeholder.merge(params.permit({ symptoms: %i[name value type label notes required] }).to_h)
-        assessment_placeholder['patient_submission_token'] = @patient_submission_token
-        # The generic 'experiencing_symptoms' boolean is used in cases where a user does not specify _which_ symptoms they are experiencing,
-        # a value of true will result in an assessment being marked as symptomatic regardless of if symptoms are specified
-        unless params.permit(:experiencing_symptoms)['experiencing_symptoms'].blank?
-          experiencing_symptoms = (%w[yes yeah].include? params.permit(:experiencing_symptoms)['experiencing_symptoms'].downcase.gsub(/\W/, ''))
-          assessment_placeholder['experiencing_symptoms'] = experiencing_symptoms
-        end
-
-        # Send the assessment to the queue for consumption
-        ProduceAssessmentJob.perform_later assessment_placeholder
-
-        # Save a new receipt and clear out any older ones
-        AssessmentReceipt.where(submission_token: @patient_submission_token).delete_all
-        assessment_receipt = AssessmentReceipt.new(submission_token: @patient_submission_token)
-        assessment_receipt.save
+      # Validate and get patient submission token and redirect if invalid link or already reported
+      @patient_submission_token = check_and_get_patient_submission_token(nil, true)
+      return if @patient_submission_token.nil?
+
+      assessment_placeholder = {}
+      assessment_placeholder = assessment_placeholder.merge(params.permit(:response_status).to_h)
+      assessment_placeholder = assessment_placeholder.merge(params.permit(:threshold_hash).to_h)
+      assessment_placeholder = assessment_placeholder.merge(params.permit({ symptoms: %i[name value type label notes required] }).to_h)
+      assessment_placeholder['patient_submission_token'] = @patient_submission_token
+      # The generic 'experiencing_symptoms' boolean is used in cases where a user does not specify _which_ symptoms they are experiencing,
+      # a value of true will result in an assessment being marked as symptomatic regardless of if symptoms are specified
+      unless params.permit(:experiencing_symptoms)['experiencing_symptoms'].blank?
+        experiencing_symptoms = (%w[yes yeah].include? params.permit(:experiencing_symptoms)['experiencing_symptoms'].downcase.gsub(/\W/, ''))
+        assessment_placeholder['experiencing_symptoms'] = experiencing_symptoms
       end
-    else
-      check_patient_token
 
+      # Send the assessment to the queue for consumption
+      ProduceAssessmentJob.perform_later assessment_placeholder
+
+      # Save a new receipt and clear out any older ones
+      AssessmentReceipt.where('BINARY submission_token = ?', @patient_submission_token).delete_all
```

Comment:
We should wipe out the AssessmentReceipts with the _old_ submission token here too

Reply:
https://github.com/SaraAlert/SaraAlert/pull/446/commits/656092432a4b0a7a4955b2cd24b40dda693ffce6